File: src/main/java/org/example/eventhub/repository/UserRepository.java
package org.example.eventhub.repository;

import java.util.Optional;
import org.example.eventhub.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;

public interface UserRepository extends JpaRepository<User, Long>, JpaSpecificationExecutor<User> {
    Optional<User> findByEmail(String email);

    Optional<User> findByUsername(String username);
}

File: src/main/kotlin/org/example/eventhub/repository/UserRepository.kt
package org.example.eventhub.repository

import org.example.eventhub.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import java.util.*

interface UserRepository : JpaRepository<User, Long>, JpaSpecificationExecutor<User> {
    fun findByEmail(email: String): Optional<User>

    fun findByUsername(username: String): Optional<User>
}
